ultralytics/data/augment.py

```diff
@@ -2011,12 +2015,62 @@ def __call__(self, labels: dict[str, Any]) -> dict[str, Any]:
                 labels["instances"].convert_bbox("xywh")
                 labels["instances"].normalize(*im.shape[:2][::-1])
                 bboxes = labels["instances"].bboxes
-                # TODO: add supports of segments and keypoints
-                new = self.transform(image=im, bboxes=bboxes, class_labels=cls)  # transformed
+                
+                # Prepare transform arguments
+                transform_args = {"image": im, "bboxes": bboxes, "class_labels": cls}
+                
+                # Add keypoints if present (for pose tasks)
+                keypoints = labels["instances"].keypoints
+                if keypoints is not None and len(keypoints):
+                    # Albumentations expects keypoints in (x, y) format, shape (N*num_kpts, 2)
+                    # YOLO keypoints are (N, num_kpts, 2 or 3) where last dim might include visibility
+                    kpts_xy = keypoints[..., :2].reshape(-1, 2)  # Flatten to (N*num_kpts, 2)
+                    keypoint_labels = [i for i in range(len(kpts_xy))]  # Create labels for each keypoint
+                    transform_args["keypoints"] = kpts_xy
+                    transform_args["keypoint_labels"] = keypoint_labels
+                
+                # Add segments if present (for segmentation tasks)
+                segments = labels["instances"].segments
+                if segments is not None and len(segments):
+                    # Albumentations expects masks as a list of polygons or binary masks
+                    # For now, we'll convert segments to masks for proper transformation
+                    # Note: This requires the image dimensions
+                    h, w = im.shape[:2]
+                    masks = [polygons2masks([seg], h, w)[0] for seg in segments] if len(segments) else []
+                    if masks:
+                        transform_args["masks"] = masks
+                
+                # Apply transformation
+                new = self.transform(**transform_args)
+                
                 if len(new["class_labels"]) > 0:  # skip update if no bbox in new im
                     labels["img"] = new["image"]
                     labels["cls"] = np.array(new["class_labels"])
                     bboxes = np.array(new["bboxes"], dtype=np.float32)
+                    
+                    # Update keypoints if they were transformed
+                    if keypoints is not None and len(keypoints) and "keypoints" in new:
+                        # Reshape back to (N, num_kpts, 2) and restore visibility if it existed
+                        num_instances = len(cls)
```

Comment:
❗ **CRITICAL**: After Albumentations runs, some bboxes may be dropped. Using `len(cls)` here can then mismatch the length of `new["keypoints"]`, causing a reshape failure. Please base the reshape on the transformed annotations (`new["class_labels"]` or `new["bboxes"]`) so the array sizes stay aligned.

**Suggested change:**
```suggestion
                        num_instances = len(new["class_labels"])
```